low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server_manifest.py

```diff
@@ -0,0 +1,335 @@
+            'input': 'hw>input[%s]>%s'
+        }
+        self.kv_dict = {}
+        sw_resources = {
+            'os': self.get_os_server_info,
+            'cortx_sw_services': self.get_cortx_service_info,
+            'external_sw_services': self.get_external_service_info
+        }
+        fw_resources = {
+            'bmc': self.get_bmc_version_info
+        }
+        self.server_resources = {
+            "fw": fw_resources,
+            "sw": sw_resources,
+            "hw": self.get_server_lshw_info
+        }
+        self.service = Service()
+        self.platform = Platform()
+
+    def get_manifest_info(self, rpath):
+        """
+        Fetch manifest information for given rpath.
+        """
+        logger.info(self.log.svc_log(
+            f"Get Manifest data for rpath:{rpath}"))
+        info = {}
+        resource_found = False
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = Utility.get_node_details(nodes[-1])
+
+        # Fetch health information for all sub nodes
+        if leaf_node == "compute":
+            info = self.get_server_manifest_info()
+            resource_found = True
+        elif leaf_node == "hw":
+            info = self.get_server_lshw_info()
+            resource_found = True
+        elif leaf_node == "sw":
+            for resource, method in self.server_resources[leaf_node].items():
+                try:
+                    info.update({resource: method()})
+                    resource_found = True
+                except Exception as err:
+                    logger.error(
+                        self.log.svc_log(f"{err.__class__.__name__}: {err}"))
+                    info = None
+        else:
+            for node in nodes:
+                resource, _ = Utility.get_node_details(node)
+                for res_type in self.server_resources:
+                    if res_type == "hw":
+                        method = self.server_resources.get(resource)
+                    else:
+                        method = self.server_resources[res_type].get(resource)
+                    if not method:
+                        logger.error(
+                            self.log.svc_log(
+                                f"No mapping function found for {res_type}"))
+                        continue
+                    try:
+                        info = method()
+                        resource_found = True
+                    except Exception as err:
+                        logger.error(
+                            self.log.svc_log(f"{err.__class__.__name__}: {err}"))
+                        info = None
+                if resource_found:
+                    break
+
+        if not resource_found:
+            msg = f"Invalid rpath or manifest provider doesn't have support for'{rpath}'."
+            logger.error(self.log.svc_log(f"{msg}"))
+            raise ManifestError(errno.EINVAL, msg)
+
+        return info
+
+    def get_server_manifest_info(self):
```

Comment:
Interface name must match the base class interface

Reply:
Done